thewalrus/decompositions.py

```diff
@@ -202,6 +203,19 @@ def takagi(A, svd_order=True):
         vals, U = takagi(Amr, svd_order=svd_order)
         return vals, U * np.exp(1j * phi / 2)
 
+    # If the matrix is diagonal, Takagi decomposition is easy
+    if np.allclose(A, np.diag(np.diag(A)), rtol=1e-16):
```

Comment:
I'd suggest you move this `rtol` as an optional parameter in the signature of the function.

Reply:
I have moved rtol as an optional parameter of takagi function.